system/perf_res_processor.py

```diff
@@ -0,0 +1,269 @@
+            print(journal_file_names)
+
+            results = []
+            ignore_list = ['grep', 'preauth', 'user']  # ignore excess entries
+            for i, file_name in enumerate(journal_file_names):
+                res = []
+                try:
+                    res += subprocess.check_output(
+                        ['xzgrep', '-i', 'exception', path + file_name]
+                    ).decode().strip().splitlines()
+                except subprocess.CalledProcessError:
+                    res += []
+                try:
+                    res += subprocess.check_output(
+                        ['xzgrep', '-i', 'error', path + file_name]
+                    ).decode().strip().splitlines()
+                except subprocess.CalledProcessError:
+                    res += []
+                res = [x for x in res if not any(map(x.__contains__, ignore_list))]
+
+                results.append(res)
+
+                # create file with exception entries for each node
+                with open(SUB_DIRS[i]+'exception_journal_entries.txt', 'w') as f:
+                    for item in res:
+                        f.write('{}\n'.format(item))
+
+            return results
+
+        for i, result in enumerate(get_journal_exceptions_as_lists(), start=1):
+            self._report.loc[[i], ['JCTL_EXCEPTIONS']] = len(result)
+
+    def process_log_errors(self, path=BASE_DIR):
+
+        def get_log_errors_as_lists():  # return list of lists
+            log_file_names = sorted(
+                [x for x in os.listdir(path) if (x.__contains__('log') and not x.__contains__('xz'))],
+                key=NATURAL_SORTING
+            )
+            print(log_file_names)
+
+            xz_file_names = sorted(
+                [x for x in os.listdir(path) if x.__contains__('xz')],
+                key=NATURAL_SORTING
+            )
+            print(xz_file_names)
+
+            node_keys = ['Node{}.'.format(i) for i in range(1, NODES_NUM+1)]
+            results = []
+            pattern_results = []
+            for i, node_key in enumerate(node_keys):
+                res = []
+                pattern_res = []
+
+                # find errors in all .log
+                log_names = [x for x in log_file_names if x.__contains__(node_key)]
+                if log_names:
+                    for log_name in log_names:
+                        try:
+                            res += subprocess.check_output(
+                                ['egrep', 'ERROR', path + log_name]
+                            ).decode().strip().splitlines()
+                        except subprocess.CalledProcessError:
+                            res += []
+
+                        # grep patterns and add them too
+                        for item in LOG_PATTERNS:
+                            try:
+                                pattern_res += subprocess.check_output(
+                                    ['egrep', item, path + log_name]
```

Comment:
i think, that xzgrep it's enough for finding

Reply:
Done.